pyproject.toml

```diff
@@ -223,3 +223,26 @@ omit = [
 ]
 [tool.coverage.run]
 relative_files = true
+
+[tool.mypy]
+# Specify the target platform details in config, so your developers are
+# free to run mypy on Windows, Linux, or macOS and get consistent
+# results.
+python_version = "3.8"
+mypy_path = "src"
+strict = true
+# Unreachable blocks have been an issue when compiling mypyc, let's try to avoid 'em in the first place.
+warn_unreachable = true
+implicit_reexport = true
+show_error_codes = true
+show_column_numbers = true
+
+[[tool.mypy.overrides]]
+module = ["pathspec.*", "IPython.*", "colorama.*", "tokenize_rt.*", "uvloop.*", "_black_version.*"]
+ignore_missing_imports = true
+
+# CI only checks src/, but in case users are running LSP or similar we explicitly ignore
+# errors in test data files.
+[[tool.mypy.overrides]]
+module = ["tests.data.*", "scripts.*"]
```

Comment:
We shouldn't ignore errors in scripts. It's fine to have a looser config apply to scripts though, like ignore missing imports

Reply:
Good point, I thought about it but then skipped as it'll add few extra ignores in repo.

- Added `type: ignore` where necessary in `scripts.*`
-  Added `types-commonmark`, `urllib3`, `hypothesmith` as `additional_dependencies` in mypy hook to avoid adding `type: ignore` (let me know if its not the right way)